### PR TITLE
[GEOT-7307] Calling getSchema on GeoJSONDataStore gives NullPointerException

### PR DESCRIPTION
**Pull request by roarbra**
[GEOT-7307](https://osgeo-org.atlassian.net/browse/GEOT-7307) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

Make sure typeName is set before using it.

# Checklist

- [X] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [X] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [X] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [X] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [X] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [X] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [X] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [X] Bug fixes and small new features are presented as a single commit.
- [X] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by mprins** [comment]
Assuming you added `empty.json` irt. the discussion in https://sourceforge.net/p/geotools/mailman/geotools-gt2-users/thread/CAHkAV%2BS72d1CSBhe%3Dj6VhmNhu%2BrG17EJg1_d0fJSvsVGNQr22g%40mail.gmail.com/#msg37774724

**Review by mprins — APPROVED** [review]

**Comment by roarbra** [comment]
Rebase seemed to work.

**Review by jodygarnett — CHANGES_REQUESTED** [review]
This class should not be caching values alongside ContentEntry. 

If additional state is needed make a subclass of GeoJSONContentEntry subclass of ContentEntry.

File: modules/unsupported/geojson-store/src/main/java/org/geotools/data/geojson/store/GeoJSONDataStore.java [hunk 1]
```diff
@@ -108,6 +108,9 @@ public void createSchema(SimpleFeatureType featureType) throws IOException {
     @Override
     public SimpleFeatureType getSchema() throws IOException {
         if (schema == null) {
+            if (typeName == null) {
+                createTypeNames();
+            }
```
[the remark this answers is not in the document]
Comment (roarbra) [reply modules/unsupported/geojson-store/src/main/java/org/geotools/data/geojson/store/GeoJSONDataStore.java:113]: @jodygarnett This PR tries to cover up for the NullPointerException. Not for the faulty implementation.

**Pull request closed without merging**